[WIP] Fix #8450: Customizing "Add Item" button text.

iamprayush opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->
Fixes #8450: Added functionality to customize the Add Item button text for the Set Input interaction to make things more apparent.
Also added a state migration job.

## Checklist
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The linter/Karma presubmit checks have passed.
  - These should run automatically, but if not, you can manually trigger them locally using `python -m scripts.pre_commit_linter` and `python -m scripts.run_frontend_tests`.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR has an appropriate "PR CHANGELOG: ..." label (If you are unsure of which label to add, ask the reviewers for guidance).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide).
- [x] The PR addresses the points mentioned in the codeowner checks for the files/folders changed. (See the [codeowner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).)
- [x] The PR is **assigned** to an appropriate reviewer.
  - If you're a new contributor, please ask on [Gitter](https://gitter.im/oppia/oppia-chat) for someone to assign a reviewer and don't tick this checkbox.
  - If you're not sure who the appropriate reviewer is, please assign to the issue's "owner" -- see the "talk-to" label on the issue. Do not only request the review but also add the reviewer as an assignee.

iamprayush committed:
Added functionality to customize the add item button. Also added a state migration job.

iamprayush commented:
This is how customizable interaction looks now:

![AddItem](https://user-images.githubusercontent.com/35144226/73777851-3196bd00-47b0-11ea-8c44-4a8f89562efe.gif)

iamprayush committed:
Fixed backend tests.

seanlip commented on core/domain/exp_domain.py:
```diff
@@ -2332,6 +2332,31 @@ def _convert_states_v29_dict_to_v30_dict(cls, states_dict):
+            states_dict: dict. A dict where each key-value pair represents,
+                respectively, a state name and a dict used to initialize a
+                State domain object.
+        Returns:
```
Add newline above this one.

seanlip commented on core/domain/question_jobs_one_off_test.py:
```diff
@@ -123,12 +123,12 @@ def test_migration_job_converts_old_question(self):
         """
-        # Generate question with old(v27) state data.
-        self.save_new_question_with_state_data_schema_v27(
+        # Generate question with old(v31) state data.
```
Not sure you should be changing this? Please read the context of the function before you change things.

seanlip commented on core/tests/test_utils.py:
```diff
@@ -249,7 +249,7 @@ class TestBase(unittest.TestCase):
     }
 
-    VERSION_27_STATE_DICT = {
+    VERSION_31_STATE_DICT = {
```
This should not be changed. Why are you changing it?

Ditto for a bunch of "27" cases below. The whole point of some of the existing tests is to ensure that entities are upgraded smoothly on-the-fly.

seanlip commented on extensions/interactions/SetInput/directives/oppia-interactive-set-input.directive.ts:
```diff
@@ -88,6 +91,10 @@ angular.module('oppia').directive('oppiaInteractiveSetInput', [
               }
             };
+            if (ctrl.buttonText) {
+              ctrl.schema.ui_config.add_element_text = $translate.instant(
```
Have you checked that this works? How did you test it?

seanlip commented:
@iamprayush Please do not submit WIP PRs. They block the Travis queue, and confuse reviewers, so I'm going to close this one.

If you need to set up your own local Travis/CircleCI instance to test a draft PR, please see the instructions at https://github.com/oppia/oppia/wiki/Setup-your-own-Travis-instance and https://github.com/oppia/oppia/wiki/Setup-your-own-CircleCI-instance. Thanks!

iamprayush commented:
> @iamprayush Please do not submit WIP PRs. They block the Travis queue, and confuse reviewers, so I'm going to close this one.
> 
> If you need to set up your own local Travis/CircleCI instance to test a draft PR, please see the instructions at https://github.com/oppia/oppia/wiki/Setup-your-own-Travis-instance and https://github.com/oppia/oppia/wiki/Setup-your-own-CircleCI-instance. Thanks!

@seanlip I added the "WIP" tag to signify that this is not ready for review, but I'll avoid it in the future, sorry about that.
Also, yes, I realize that they block Travis queue which is why I canceled the build as soon as I added the tag. I shall create a new PR after addressing the changes you mentioned and fixing the failing tests. Thanks!
